generate.sh

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env bash
+
+usage()
+{
+  echo "Common settings:"
+  echo "  --platform <value>         Platform: MacOSX, iPhoneOS, iPhoneSimulator, AppleTVOS, AppleTVSimulator"
+  echo "  --arch <value>             Architecture: arm64e-apple-macos, x86_64-apple-macos"
+  echo "  --framework                Framework to generate bindings for"
+  echo "  --help                     Print help and exit (short: -h)"
+  echo ""
+
+  echo "Actions:"
+  echo "  --experimental             Generates only Runtime.Swift namespace when bindings for frameworks are not complete"
+}
+
+source="${BASH_SOURCE[0]}"
+
+# resolve $SOURCE until the file is no longer a symlink
+while [[ -h $source ]]; do
+  scriptroot="$( cd -P "$( dirname "$source" )" && pwd )"
+  source="$(readlink "$source")"
+
+  # if $source was a relative symlink, we need to resolve it relative to the path where the
+  # symlink file was located
+  [[ $source != /* ]] && source="$scriptroot/$source"
+done
+
+scriptroot="$( cd -P "$( dirname "$source" )" && pwd )"
+
+platform=''
+arch=''
+frameworks=()
+experimental=false
+
+output_dir="./GeneratedBindings"
+
+while [[ $# > 0 ]]; do
+  opt="$(echo "${1/#--/-}" | tr "[:upper:]" "[:lower:]")"
+  case "$opt" in
+    -help|-h)
+      usage
+      exit 0
+      ;;
+    -experimental)
+      experimental=true
+      ;;
+    -platform)
+      platform=$2
+      shift
+      ;;
+    -arch)
+      arch=$2
+      shift
+      ;;
+    -framework)
+      frameworks+=("$2")
+      shift
+      ;;
+  esac
+
+  shift
+done
+
+# Output directory for generated bindings
+rm -rf "$output_dir"
+mkdir -p "$output_dir"
+
+cd "$output_dir"
+
+# Function to extract ABI file
+function ExtractABI {
+    local framework=$1
+
+    echo "Generating ABI for framework '$framework', platform '$platform', architecture '$arch'"
+    
+    local sdk_path=$(xcrun -sdk $(echo "$platform" | tr '[:upper:]' '[:lower:]') --show-sdk-path)
+    local swift_interface_path="/Applications/Xcode.app/Contents/Developer/Platforms/${platform}.platform/Developer/SDKs/${platform}.sdk/System/Library/Frameworks/${framework}.framework/Versions/Current/Modules/${framework}.swiftmodule/${arch}.swiftinterface"
```

Comment:
nit: use `xcode-select -p` to figure out path to Xcode rather than hardcoding /Applications